c++/src/kj/compat/tls.c++

```diff
@@ -100,6 +100,37 @@ inline void ensureOpenSslInitialized() {
 }
 #endif
 
+bool isIpAddress(kj::StringPtr addr) {
+  bool isPossiblyIp6 = true;
+  bool isPossiblyIp4 = true;
+  uint colonCount = 0;
+  uint dotCount = 0;
+  for (auto c: addr) {
+    if (c == ':') {
+      isPossiblyIp4 = false;
+      ++colonCount;
+    } else if (c == '.') {
+      isPossiblyIp6 = false;
+      ++dotCount;
+    } else if ('0' <= c && c <= '9') {
+      // Digit is valid for ipv4 or ipv6.
+    } else if ('a' <= c && c <= 'f') {
+      // Hex digit could be ipv6 but not ipv4.
+      isPossiblyIp4 = false;
+    } else {
+      // Nope.
+      return false;
+    }
+  }
+
+  // An IPv4 address has 3 dots. (Yes, I'm aware that technically IPv4 addresses can be formatted
+  // with fewer dots, but it's not clear that we actually want to support TLS authentication of
+  // non-canonical address formats, so for now I'm not. File a bug if you care.) An IPv6 address
+  // has at least 2 and as many as 7 colons.
+  return (isPossiblyIp4 && dotCount == 3)
```

Comment:
I've been considering the possibility of moving the whatwg url parsing algorithm into kj as part of kj::Url... in which case we can likely use it's ip parsing algorithms to support those other formats if we wanted later. I agree for now that this is more than sufficient.

Reply:
I am not sure I want the whatwg algorithm in `kj::Url`. I prefer that `kj::Url` decodes %-escapes correctly. :)

Also note that `kj::Network::parseAddress()` actually does parse any IP address format correctly, using `inet_pton()`, but as the comment says it's not clear to me we actually want to support weird formats here.

Reply:
Fair enough!